packages/appium/lib/extension/extension-config.js

```diff
@@ -0,0 +1,458 @@
+// @ts-check
+
+import _ from 'lodash';
+import path from 'path';
+import resolveFrom from 'resolve-from';
+import log from '../logger';
+import {
+  ALLOWED_SCHEMA_EXTENSIONS,
+  isAllowedSchemaFileExtension,
+  registerSchema,
+} from '../schema/schema';
+
+const INSTALL_TYPE_NPM = 'npm';
+const INSTALL_TYPE_LOCAL = 'local';
+const INSTALL_TYPE_GITHUB = 'github';
+const INSTALL_TYPE_GIT = 'git';
+
+/** @type {Set<InstallType>} */
+const INSTALL_TYPES = new Set([
+  INSTALL_TYPE_GIT,
+  INSTALL_TYPE_GITHUB,
+  INSTALL_TYPE_LOCAL,
+  INSTALL_TYPE_NPM,
+]);
+
+/**
+ * This class is abstract. It should not be instantiated directly.
+ *
+ * Subclasses should provide the generic parameter to implement.
+ * @template {ExtensionType} ExtType
+ */
+export class ExtensionConfig {
+  /** @type {ExtType} */
+  extensionType;
+
+  /** @type {`${ExtType}s`} */
+  configKey;
+
+  /** @type {ExtRecord<ExtType>} */
+  installedExtensions;
+
+  /** @type {ExtensionLogFn} */
+  log;
+
+  /** @type {import('./manifest').Manifest} */
+  io;
```

Comment:
is there any reason to call this variable `io`? I cannot really understand what it contains

Reply:
Yes, that is old. I'll change it.